src/asset_manager/asset_manager.rs

```diff
@@ -410,59 +422,67 @@ impl AssetLoader<Mesh> for ObjSet {
         // flat vec of `VertexPosNormal` objects.
         // TODO: Doesn't differentiate between objects in a `*.obj` file, treats
         // them all as a single mesh.
-        let vertices: Vec<VertexPosNormal> = obj_set.objects
+        let vertices: Vec<VertexPosNormal> = obj_set
+            .objects
             .iter()
             .flat_map(|object| {
-                object.geometry
+                object
+                    .geometry
                     .iter()
                     .flat_map(|ref geometry| {
-                        geometry.shapes.iter().flat_map(|s| -> Vec<VertexPosNormal> {
-                            let mut vtn_indices = vec![];
-
-                            match s.primitive {
-                                Primitive::Point(v1) => {
-                                    vtn_indices.push(v1);
-                                }
-                                Primitive::Line(v1, v2) => {
-                                    vtn_indices.push(v1);
-                                    vtn_indices.push(v2);
-                                }
-                                Primitive::Triangle(v1, v2, v3) => {
-                                    vtn_indices.push(v1);
-                                    vtn_indices.push(v2);
-                                    vtn_indices.push(v3);
-                                }
-                            }
-
-                            vtn_indices.iter()
-                                .map(|&(vi, ti, ni)| {
-                                    let vertex = object.vertices[vi];
-
-                                    VertexPosNormal {
-                                        pos: [vertex.x as f32, vertex.y as f32, vertex.z as f32],
-                                        normal: match ni {
-                                            Some(i) => {
-                                                let normal = object.normals[i];
-
-                                                Vector3::from([normal.x as f32,
-                                                               normal.y as f32,
-                                                               normal.z as f32])
-                                                    .normalize()
-                                                    .into()
-                                            }
-                                            None => [0.0, 0.0, 0.0],
-                                        },
-                                        tex_coord: match ti {
-                                            Some(i) => {
-                                                let tvertex = object.tex_vertices[i];
-                                                [tvertex.u as f32, tvertex.v as f32]
-                                            }
-                                            None => [0.0, 0.0],
-                                        },
+                        geometry
```

Comment:
I don't think mixing your input PR with rustfmt pass is a good idea. Better have them separate. FWIW, I like the old formatting better.